File: backend/app/routers/jobs.py
from __future__ import annotations

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app import crud
from app.deps import get_current_user
from app.db import get_db
from app import models
from app.schemas import JobCreate, JobOut, JobUpdate


router = APIRouter(prefix='/jobs', tags=['jobs'])


@router.get('', response_model=list[JobOut])
def list_jobs(db: Session = Depends(get_db), _current_user: models.User = Depends(get_current_user)):
  return crud.list_jobs(db)


@router.post('', response_model=JobOut)
def create_job(data: JobCreate, db: Session = Depends(get_db), _current_user: models.User = Depends(get_current_user)):
  return crud.create_job(db, data)


@router.get('/{job_id}', response_model=JobOut)
def get_job(job_id: int, db: Session = Depends(get_db), _current_user: models.User = Depends(get_current_user)):
  job = crud.get_job(db, job_id)
  if not job:
    raise HTTPException(status_code=404, detail='Job not found')
  return job


@router.put('/{job_id}', response_model=JobOut)
def update_job(job_id: int, data: JobUpdate, db: Session = Depends(get_db), _current_user: models.User = Depends(get_current_user)):
  job = crud.get_job(db, job_id)
  if not job:
    raise HTTPException(status_code=404, detail='Job not found')
  return crud.update_job(db, job, data)


@router.delete('/{job_id}')
def delete_job(job_id: int, db: Session = Depends(get_db), _current_user: models.User = Depends(get_current_user)):
  job = crud.get_job(db, job_id)
  if not job:
    raise HTTPException(status_code=404, detail='Job not found')
  crud.delete_job(db, job)
  return {'ok': True}

File: src/renderer/src/api/jobs.ts
import { request } from './http'
import type { Job } from './types'

export type JobCreate = Omit<Job, 'id' | 'createdAt'>
export type JobUpdate = Partial<JobCreate>

export function listJobs(): Promise<Job[]> {
  return request<Job[]>('/jobs')
}

export function getJob(jobId: number): Promise<Job> {
  return request<Job>(`/jobs/${jobId}`)
}

export function createJob(data: JobCreate): Promise<Job> {
  return request<Job>('/jobs', { method: 'POST', body: data })
}

export function updateJob(jobId: number, data: JobUpdate): Promise<Job> {
  return request<Job>(`/jobs/${jobId}`, { method: 'PUT', body: data })
}

export function deleteJob(jobId: number): Promise<{ ok: boolean }> {
  return request<{ ok: boolean }>(`/jobs/${jobId}`, { method: 'DELETE' })
}
